hack/ci/install-cnv.sh

```diff
@@ -2,12 +2,45 @@
 
 set -ex
 
+CNV_VERSION=4.12
+
 # The kubevirt tests require wildcard routes to be allowed
 oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'
 
 # Make the masters schedulable so we have more capacity to run VMs
 oc patch scheduler cluster --type=json -p '[{ "op": "replace", "path": "/spec/mastersSchedulable", "value": true }]'
 
+# Add pullsecret for cnv nightly channel from quay.io/openshift-cnv
+QUAY_USERNAME=openshift-cnv+openshift_ci
+QUAY_PASSWORD=$(jq -r .openshift_cnv_pullsecret /etc/cnv-nightly-pull-credentials)
+oc get secret pull-secret -n openshift-config -o json | jq -r '.data.".dockerconfigjson"' | base64 -d > global-pull-secret.json
+QUAY_AUTH=$(echo -n "${QUAY_USERNAME}:${QUAY_PASSWORD}" | base64 -w 0)
+jq --arg QUAY_AUTH "$QUAY_AUTH" '.auths += {"quay.io/openshift-cnv": {"auth":$QUAY_AUTH,"email":""}}' global-pull-secret.json > global-pull-secret.json.tmp
+mv global-pull-secret.json.tmp global-pull-secret.json
+oc set data secret/pull-secret -n openshift-config --from-file=.dockerconfigjson=global-pull-secret.json
+rm global-pull-secret.json
+
+sleep 5
+
+oc wait mcp master worker --for condition=updated --timeout=20m
+
+# Create a catalog source for the pre-release builds
+cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: cnv-nightly-catalog-source
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/openshift-cnv/nightly-catalog:${CNV_VERSION}
+  displayName: OpenShift Virtualization Nightly Index
+  publisher: Red Hat
+  updateStrategy:
+    registryPoll:
+      interval: 8h
+EOF
+
```

Comment:
based on my CNV_PRERELEASE_VERSION env var suggestion, this whole section would only be called if CNV_PRERELEASE_VERSION is set.

Reply:
done.